rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -200,8 +192,10 @@ object Reduce {
           case _ => s.unit
         }
 
-      def mkJob[A](terms : Seq[A], handler : (A, Blake2b512Random) => M[Unit]) : M[List[Unit]] =
-        Parallel.parTraverse(terms.zipWithIndex.toList)(handle(handler))
+      def mkJob[A](terms: Seq[A], handler: (A, Blake2b512Random) => M[Unit]): M[List[Unit]] =
+        Parallel.parTraverse(terms.zipWithIndex.toList) {
+          case (term, idx) => handler(term, split(rand, idx))
```

Comment:
can we make it a total funciton using `Function.tupled`?

Reply:
This is function was abandoned during the refactoring but thanks for telling me about Function.tupled.  I didn't know about its existence.